### PR TITLE
[CCT-432]feat(foreground notifications)Create component

### PR DESCRIPTION
**Pull request by calcerrada**
### Documentation
- [x] Add the issue in PR description
- [x] Have changelog updated
- [x] You fill the milestone value
- [x] You add some label
- [x]  Have example running in demo app
- [x] Review id on demo is the same of egeo folder

### Code
- [ ] Pass Sass lint task
- [ ] Have qaTags

### Tests
- [x] Have at least 70% tests coverage

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +15,7 @@
 * st-header: Add new parameter "openInNewPage" to model to define if open new tab when navigate to link 
 * st-progress-bar: create component
 * st-widget: Implement loading state
-
+* st-foreground-notifications: create component
```
Comment (ppenalver) [inline CHANGELOG.md]: white line under this line

File: src/egeo-demo/egeo-demo.routes.ts [hunk 2]
```diff
@@ -32,6 +32,8 @@ export const EGEO_DEMO_MENU: EgeoDemoMenu[] = [
    { id: 'st-dropdown-menu', name: 'Dropdown menu', path: 'dropdown-menu-demo', moduleLazyLoad: 'StDropdownMenuDemoModule' },
    { id: 'st-file-button', name: 'File button', path: 'file-button-demo', moduleLazyLoad: 'StFileButtonDemoModule' },
    { id: 'st-footer', name: 'Footer', path: 'footer-demo', moduleLazyLoad: 'StFoterDemoModule' },
+   { id: 'st-foreground-notifications', name: 'Foreground-notifications',
+      path: 'foreground-notifications-demo', moduleLazyLoad: '@StForegroundNotificationsModule' },
```
Comment (ppenalver) [inline src/egeo-demo/egeo-demo.routes.ts]: indent

File: src/egeo-demo/st-foreground-notifications-demo/st-foreground-notifications-demo.component.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   templateUrl: './st-foreground-notifications-demo.component.html',
+   styleUrls: ['./st-foreground-notifications-demo.component.scss']
+})
+export class StForegroundNotificationsDemoComponent {
+   isVisible: boolean = true;
+   items: any[]= [
+      {
+         text : 'This is a neutral informational notification',
+         status: '',
+         viisible: true
```
Comment (ppenalver) [inline src/egeo-demo/st-foreground-notifications-demo/st-foreground-notifications-demo.component.ts]: vii

File: src/lib/st-foreground-notifications/st-foreground-notifications.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+   /** @Input {NotificationStatus} [status='NotificationStatus.default'] Defines the criticality level of the notification */
+   @Input() status: string = 'default';
+
+
+   public hide: boolean;
+   public fadeOut: boolean;
+   private _visible: boolean;
+   private _blocked: boolean;
+
+   constructor(private cd: ChangeDetectorRef, private el: ElementRef) { }
```
Comment (ppenalver) [inline src/lib/st-foreground-notifications/st-foreground-notifications.ts]: element ref its not necesary

File: src/lib/st-foreground-notifications/st-foreground-notifications.spec.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+         expect(nativeElement.querySelector('.foregrond-notification').classList).not.toContain('success');
+         expect(nativeElement.querySelector('.foregrond-notification').classList).not.toContain('default');
+      }));
+
+      it('And status is not defined, element should be contain "default" class', async(() => {
+         comp.visible = true;
+         fixture.detectChanges();
+
+         expect(nativeElement.querySelector('st-foreground-notifications').classList).toContain('visible');
+         expect(nativeElement.querySelector('.foregrond-notification').classList).toContain('default');
```
Comment (EvaMGC) [inline src/lib/st-foreground-notifications/st-foreground-notifications.spec.ts]: Change 'foregrond' to 'foreground'

File: src/lib/st-foreground-notifications/st-foreground-notifications.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+   host: {
+      '[class.visible]': 'visible',
+      '[class.fadeout]': 'fadeOut',
+      '[class.hide]': 'hide'
+   }
+})
+/**
+ * @description {Component} [Foreground notifications]
+ *
+ * Foreground notifications are made to let the user know info about a process she is performing in real time.
```
Comment (EvaMGC) [inline src/lib/st-foreground-notifications/st-foreground-notifications.ts]: Change "are made to let the user know info about a process she is performing in real time" to
"are used for letting user know feedback about a process executed in real time"

File: src/lib/st-foreground-notifications/st-foreground-notifications.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+   @Input()
+   public get visible(): boolean {
+      return this._visible;
+   }
+
+   public set visible(value: boolean) {
+      this._visible = value;
+      this.fadeOut = !value;
+      setTimeout(() => {
+         this.hide = !this._visible;
```
Comment (EvaMGC) [inline src/lib/st-foreground-notifications/st-foreground-notifications.ts]: Why do you need two variables (visible and hide)?

File: src/lib/st-foreground-notifications/st-foreground-notifications.spec.ts [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+         expect(nativeElement.querySelector('st-foreground-notifications').classList).toContain('visible');
+         expect(nativeElement.querySelector('.foreground-notification').classList).toContain('success');
+         expect(nativeElement.querySelector('.foreground-notification').classList).not.toContain('warning');
+         expect(nativeElement.querySelector('.foreground-notification').classList).not.toContain('default');
+         expect(nativeElement.querySelector('.foreground-notification').classList).not.toContain('critical');
+
+
+      }));
+
+      it('And status is defined as default, element should be contain "warning" class', async(() => {
```
Comment (EvaMGC) [inline src/lib/st-foreground-notifications/st-foreground-notifications.spec.ts]: Change 'should be contain' to 'should contain'

**Pull request merged**